filesystem.go

```diff
@@ -1,92 +1,148 @@
 package function
 
 import (
+	"archive/zip"
+	"bytes"
 	"io"
+	"io/fs"
 	"os"
+	"path"
 	"path/filepath"
 	"sort"
+	"strings"
 
 	billy "github.com/go-git/go-billy/v5"
-	"github.com/markbates/pkger"
 )
 
 // Filesystems
 // Wrap the implementations of FS with their subtle differences into the
 // common interface for accessing template files defined herein.
 // os:    standard for on-disk extensible template repositories.
-// pker:  embedded filesystem backed by the generated pkged.go.
+// zip:   embedded filesystem backed by the byte array representing zipfile.
 // billy: go-git library's filesystem used for remote git template repos.
 
 type Filesystem interface {
-	Stat(name string) (os.FileInfo, error)
-	Open(path string) (file, error)
-	ReadDir(path string) ([]os.FileInfo, error)
+	fs.ReadDirFS
+	fs.StatFS
 }
 
-type file interface {
-	io.Reader
-	io.Closer
+type zipFS struct {
+	archive *zip.Reader
 }
 
-// pkgerFilesystem is template file accessor backed by the pkger-provided
-// embedded filesystem.o
-type pkgerFilesystem struct{}
-
-// the root of the repository is actually ./templates, which is proffered
-// in the pkger filesystem as /templates, so all path requests will be
-// prefixed with this path to emulate having the pkger fs root the same
-// as the logical root.
-const pkgerRoot = "/templates"
-
-func (a pkgerFilesystem) Stat(path string) (os.FileInfo, error) {
-	return pkger.Stat(filepath.Join(pkgerRoot, path))
+func (z zipFS) Open(name string) (fs.File, error) {
+	return z.archive.Open(name)
 }
 
-func (a pkgerFilesystem) Open(path string) (file, error) {
-	return pkger.Open(filepath.Join(pkgerRoot, path))
+func (z zipFS) ReadDir(name string) ([]fs.DirEntry, error) {
+	var dirEntries []fs.DirEntry
+	for _, file := range z.archive.File {
+		cleanName := strings.TrimRight(file.Name, "/")
+		if path.Dir(cleanName) == name {
+			f, err := z.archive.Open(cleanName)
+			if err != nil {
+				return nil, err
+			}
+			fi, err := f.Stat()
+			if err != nil {
+				return nil, err
+			}
+			dirEntries = append(dirEntries, dirEntry{FileInfo: fi})
+		}
+	}
+	return dirEntries, nil
 }
 
-func (a pkgerFilesystem) ReadDir(path string) ([]os.FileInfo, error) {
-	f, err := pkger.Open(filepath.Join(pkgerRoot, path))
+func (z zipFS) Stat(name string) (fs.FileInfo, error) {
+	f, err := z.archive.Open(name)
 	if err != nil {
 		return nil, err
 	}
-	return f.Readdir(-1)
+	return f.Stat()
 }
 
+//go:generate go run ./generate/templates/main.go
```

Comment:
It generates variable `templatesZip` that is used here, just few lines below.

Reply:
>//go:generate go run ./generate/templates/main.go

Could be probably in any file but I put it here since the product of the generation (the byte array) is used here.